src/view_translate_bisulfite.hpp

```diff
@@ -0,0 +1,428 @@
+//         return urange[index_urange] | seqan3::views::to_bisulfite_semialphabet(index_strand - 1); // We want forward strand first
+//     }
+//
+//     const_reference operator[](size_type const n) const
+//         requires seqan3::const_iterable_range<urng_t>
+//     {
+//         assert(n < size());
+//         size_type index_strand = n % 2;
+//         size_type index_urange = (n - index_strand) / 2;
+//         return urange[index_urange] | seqan3::views::to_bisulfite_semialphabet(index_strand - 1);
+//     }
+// };
+//
+// template <typename urng_t>
+// view_translate_bisulfite(urng_t &&) -> view_translate_bisulfite<std::ranges::all_view<urng_t>>;
+//
+// struct translate_bisulfite_fn
+// {
+//     template <std::ranges::range urng_t>
+//     constexpr auto operator()(urng_t && urange) const
+//     {
+//         static_assert(seqan3::dimension_v<urng_t> == 2,
+//                       "This adaptor only handles range-of-range (two dimensions) as input.");
+//         static_assert(std::ranges::viewable_range<urng_t>,
+//                       "The range parameter to views::translate_bisulfite cannot be a temporary of a non-view range.");
+//         static_assert(std::ranges::viewable_range<std::ranges::range_reference_t<urng_t>>,
+//                       "The inner range of the range parameter to views::translate_bisulfite cannot be a "
+//                       "temporary of a non-view range.");
+//         static_assert(std::ranges::sized_range<urng_t>,
+//                       "The range parameter to views::translate_bisulfite must model std::ranges::sized_range.");
+//         static_assert(std::ranges::sized_range<std::ranges::range_reference_t<urng_t>>,
+//                       "The inner range of the range parameter to views::translate_bisulfite must model "
+//                       "std::ranges::sized_range.");
+//         static_assert(std::ranges::random_access_range<urng_t>,
+//                       "The range parameter to views::translate_bisulfite must model std::ranges::random_access_range.");
+//         static_assert(std::ranges::random_access_range<std::ranges::range_reference_t<urng_t>>,
+//                       "The inner range of the range parameter to views::translate_bisulfite must model "
+//                       "std::ranges::random_access_range.");
+//         static_assert(std::is_same_v<std::remove_const_t<seqan3::innermost_value_type_t<urng_t>>, seqan3::dna4>,
+//                       "The range parameter to views::translate_bisulfite must be over a range over elements of "
+//                       "seqan3::dna4.");
+//
+//         return view_translate_bisulfite{std::forward<urng_t>(urange)};
+//     }
+//
+//     template <std::ranges::range urng_t>
+//     constexpr friend auto operator|(urng_t && urange, translate_bisulfite_fn const & me)
+//     {
+//         return me(std::forward<urng_t>(urange));
+//     }
+// };
+
+struct translate_bisulfite_fn
```

Comment:
Maybe 
```suggestion
struct reduce_to_bisulfite_fn
```
?

We are not doing translation and the number of frames doesn't increase in this step anymore.